page-objects/pages/TASKListPage.js

```diff
@@ -160,67 +147,74 @@ async function clickDeleteTaskButtonInPopup() {
+	if (clickOnThatTask === undefined){
+		let isTaskClickable = new Boolean (false);
+		while (!isTaskClickable){
+			elementHelpers.scrollToElement(clickOnThatTask);
+			await waitHelpers.waitUntilElementIsVisible(clickOnThatTask);
+			if (clickOnThatTask.isDisplayedInViewport()){
+				isTaskClickable = true;
+			}
+		}
+	}
+	if (clickOnThatTask.isDisplayedInViewport()) {
+		await elementHelpers.clickAndWait(clickOnThatTask);
+	}else{
+		await elementHelpers.scrollToElement(clickOnThatTask);
+		await elementHelpers.clickAndWait(clickOnThatTask);
+	}
 }
 
-async function getTaskFromNuxtClient(taskName) {
-	await driver.pause(3000);
-	await waitHelpers.waitUntilElementIsVisible(taskOverviewLoad);
-	await driver.pause(3000);
-	const taskOverviewResult = await getNuxtTaskList();
-	const taskIndex = taskOverviewResult.indexOf(taskName);
-	let clickOnTask = taskOverviewResult[taskIndex];
-	const taskInTheList = '//div[text() =' + "'" + clickOnTask + "'" + ']';
-	return taskInTheList;
+async function getTaskFromTaskOverview(taskName) {
+	let tasksOnThePage = [];
+	await waitHelpers.waitUntilNuxtClientLoads();
+	await waitHelpers.waitUntilElementIsPresent(taskSection);
+	await driver.$(taskSection).$$(taskTitle).forEach(async function (element)  {
+		tasksOnThePage.push(await element.getText());
+	})
+	let isTaskInTheList = new Boolean(false);
+	isTaskInTheList =  (tasksOnThePage.includes(taskName)) ? driver.$(mod_extsprintf.sprintf(taskTitleText, taskName)) : false ;
+	return isTaskInTheList
 }
 
 async function taskDisplayed(taskName) {
-	let taskInTheList = (await getTaskFromNuxtClient(taskName)).toString();
+	let taskInTheList = (await getTaskFromTaskOverview(taskName));
 	await waitHelpers.waitUntilElementIsPresent(taskInTheList);
 }
 
 async function taskNotDisplayed(taskName) {
-	let taskInTheList = (await getTaskFromNuxtClient(taskName)).toString();
-	await waitHelpers.waitUntilElementIsNotPresent(taskInTheList);
+	let taskInTheList = (await getTaskFromTaskOverview(taskName));
+	expect(taskInTheList).to.equal(false);
 }
 
 async function studentSubmittedTask() {
 	await elementHelpers.clickAndWait(submittedTask);
 	await elementHelpers.clickAndWait(studentSubmitTask);
 }
 
-async function getNuxtTaskList() {
-	const listOfAllNuxtTasks = [];
-	let elements = await driver.$(taskSection);
-	await elements.$$(taskTitle).map(async function (element) {
-		listOfAllNuxtTasks.push(await element.getText());
-	});
-	return listOfAllNuxtTasks;
-}
-
 async function hoverOverTaskAndClickMenu(taskName) {
-	await driver.pause(5000);
 	let taskTitle = await driver.$(mod_extsprintf.sprintf(taskTitleText, taskName));
-	await taskTitle.scrollIntoView(false);
-	if (await taskTitle.isDisplayedInViewport()){
-		let xOffset = await taskTitle.getLocation('x');
-		let yOffset = await taskTitle.getLocation('y');
-		taskTitle.moveTo(xOffset, yOffset);
-		await driver.pause(5000);
+	await waitHelpers.waitUntilNuxtClientLoads();
```

Comment:
I think we should exchange the wait above with this one 'await waitHelpers.waitUntilElementIsPresent(taskSection);'.